### PR TITLE
[ci] Collect and replay skipped tests

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

This PR is targetted at speeding up our tests on appveyor by skipping
already succeeding tests when enabling addtional extensions.

Once it's confirmed it works, I'll port it to other branches.

### CODE REVIEW

File: phpunit [hunk 1]
```diff
@@ -41,7 +41,16 @@ if (!file_exists("$PHPUNIT_DIR/phpunit-$PHPUNIT_VERSION/phpunit")) {
     $zip->close();
     chdir("phpunit-$PHPUNIT_VERSION");
     passthru("$COMPOSER remove --no-update symfony/yaml");
+    passthru("$COMPOSER require --no-update symfony/phpunit-bridge ~2.8");
```
Comment (stof) [inline phpunit]: why ? The Symfony project already depend on the bridge

Comment (nicolas-grekas) [inline phpunit]: This is phpunit's composer.json

Comment (nicolas-grekas) [inline phpunit]: I should try removing this :)

Comment (nicolas-grekas) [inline phpunit]: Well, phpunit listeners are loaded before the bootstrapp file is included, which means we have to add phpunit-bridge to phpunit's composer.json, no choice

File: phpunit [hunk 2]
```diff
@@ -41,7 +41,9 @@ if (!file_exists("$PHPUNIT_DIR/phpunit-$PHPUNIT_VERSION/phpunit")) {
     $zip->close();
     chdir("phpunit-$PHPUNIT_VERSION");
     passthru("$COMPOSER remove --no-update symfony/yaml");
+    passthru("$COMPOSER require --dev --no-update symfony/phpunit-bridge '>=2.8@dev'");
     passthru("$COMPOSER install --prefer-source --no-progress --ansi");
+    copy('../../src/Symfony/Bridge/PhpUnit/SkippedTestsListener.php', './vendor/symfony/phpunit-bridge/SkippedTestsListener.php');
```
Comment (nicolas-grekas) [inline phpunit]: This line should be removed before merging. For now, it allows testing before subtree-splitting

File: appveyor.yml [hunk 3]
```diff
@@ -32,6 +33,7 @@ install:
     - IF %PHP%==1 cd ..
     - IF %PHP%==1 echo @php %%~dp0composer.phar %%* > composer.bat
     - IF %PHP%==1 copy /Y php.ini-development php.ini-min
+    - IF %PHP%==1 echo max_execution_time=1200 >> php.ini-min
```
Comment (nicolas-grekas) [inline appveyor.yml]: 20 minutes is twice the time required for tests to run. This prevents blocking for appveyor queue for too long.

File: phpunit [hunk 4]
```diff
@@ -41,7 +41,9 @@ if (!file_exists("$PHPUNIT_DIR/phpunit-$PHPUNIT_VERSION/phpunit")) {
     $zip->close();
     chdir("phpunit-$PHPUNIT_VERSION");
     passthru("$COMPOSER remove --no-update symfony/yaml");
+    passthru("$COMPOSER require --dev --no-update symfony/phpunit-bridge \">=2.8@dev\"");
     passthru("$COMPOSER install --prefer-source --no-progress --ansi");
+    symlink(realpath('../../src/Symfony/Bridge/PhpUnit/SkippedTestsListener.php'), './vendor/symfony/phpunit-bridge/SkippedTestsListener.php');
```
Comment (nicolas-grekas) [inline phpunit]: This line should be removed before merging. For now, it allows testing before subtree-splitting

**Comment by nicolas-grekas** [comment]
PR is ready for review, failures on Windows are not related

**Comment by nicolas-grekas** [comment]
Splitted in two PR, #16190 on 2.3 and #16189 on 2.8

**Pull request closed without merging**